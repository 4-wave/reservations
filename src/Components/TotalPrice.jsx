import React from 'react';
import { Wrapper, Container, Informational, Price } from './styles/TotalPrice.js';

class TotalPrice extends React.Component{
    constructor(props) {
        super(props)
        this.state = {
            pricePerNight: this.props.pricePerNight,
            cleaningFees: this.props.cleaningFees,
            serviceFees: this.props.serviceFees,
            amountOfNights: this.calcTotalNights()
        }
        this.calcTotalNights = this.calcTotalNights.bind(this);
    }

    componentDidUpdate() {
        if (this.state.pricePerNight !== this.props.pricePerNight ||
             this.state.cleaningFees !== this.props.cleaningFees ||
             this.state.serviceFees !== this.props.serviceFees ||
             this.props.end - this.props.start !== this.state.amountOfNights) {
                this.setState({pricePerNight: this.props.pricePerNight,
                cleaningFees: this.props.cleaningFees,
                serviceFees: this.props.serviceFees,
            amountOfNights: this.calcTotalNights()})
             }
    }

    calcTotalNights() {
        return this.props.end - this.props.start;
    }

    render() {
      let total = this.calcTotalNights();
      let totalBillForRooms = total * this.state.pricePerNight;
        return(
            <Wrapper>
        <Container><Informational>{this.state.pricePerNight} x {total}</Informational><Price>{this.state.pricePerNight * total}</Price></Container>
            <Container><Informational>Service fees</Informational><Price>{this.state.serviceFees}</Price></Container>
        <Container><Informational>Occupancy taxes and fees</Informational><Price>{this.state.cleaningFees}</Price></Container>
    <Container><Informational>Total:</Informational><Price>{this.state.cleaningFees + this.state.serviceFees + totalBillForRooms}</Price></Container>
            </Wrapper>
        )
    }
}

export default TotalPrice;