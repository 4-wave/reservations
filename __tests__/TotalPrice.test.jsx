import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { create } from "react-test-renderer";
import TotalPrice from '../src/Components/TotalPrice.jsx';

describe('<TotalPrice /> component', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<TotalPrice />);

        expect(tree).toMatchSnapshot();
    });

    it('toggles the state for night message', () => {
        const props = {pricePerNight: 0, cleaningFees: undefined,
            serviceFees: undefined, amountOfNights: NaN, end: 0, start: 0}
        const wrapper = create(<TotalPrice {...props}/>);
        const instance = wrapper.getInstance();
        console.log(instance.state)
        expect(instance.state.isNightMessageClicked).toEqual(false);
        instance.clickNightMessage();
        expect(instance.state.isNightMessageClicked).toEqual(true);
    });
});
