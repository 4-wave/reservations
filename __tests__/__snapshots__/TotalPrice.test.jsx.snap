// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TotalPrice /> component matches the snapshot 1`] = `
<div
  className="sc-bdVaJa fyIdvd"
>
  <div
    className="sc-bwzfXH fkqeJu"
  >
    <div
      className="sc-htpNat hMxhSW"
    >
       x 
      NaN
       nights 
      <button
        className="sc-ifAKCX hZwRRR"
        onClick={[Function]}
      >
        ?
      </button>
    </div>
    <div
      className="sc-bxivhb hExhWl"
    >
      NaN
    </div>
  </div>
  <div />
  <div
    className="sc-bwzfXH fkqeJu"
  >
    <div
      className="sc-htpNat hMxhSW"
    >
      Service fees 
      <button
        className="sc-ifAKCX hZwRRR"
        onClick={[Function]}
      >
        ?
      </button>
    </div>
    <div
      className="sc-bxivhb hExhWl"
    />
  </div>
  <div />
  <div
    className="sc-bwzfXH fkqeJu"
  >
    <div
      className="sc-htpNat hMxhSW"
    >
      Occupancy taxes and fees
    </div>
    <div
      className="sc-bxivhb hExhWl"
    />
  </div>
  <div
    className="sc-bwzfXH fkqeJu"
  >
    <div
      className="sc-htpNat hMxhSW"
    >
      Total:
    </div>
    <div
      className="sc-bxivhb hExhWl"
    >
      NaN
    </div>
  </div>
</div>
`;
